File: 78.subsets.py
#
# @lc app=leetcode id=78 lang=python3
#
# [78] Subsets
#

from typing import *
# @lc code=start
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        output = [[]]
        for num in nums:
            output += [curr+[num] for curr in output]
        return output
        
# @lc code=end
if __name__ == "__main__":
    print(Solution().subsets([1,2,3]))

File: golang/78.subsets.go
/*
 * @lc app=leetcode.cn id=78 lang=golang
 *
 * [78] Subsets
 */
package main

// @lc code=start
func subsets(nums []int) [][]int {
	n := 1 << len(nums)
	ret := [][]int{}
	for i := 0; i < n; i++ {
		j := i
		item := []int{}
		digit := 0
		for j > 0 {
			if j&1 == 1 {
				item = append(item, nums[digit])
			}
			digit += 1
			j >>= 1
		}
		ret = append(ret, item)
	}
	return ret
}

// @lc code=end
